tests/ContextTest.py

```diff
@@ -95,5 +97,103 @@ def test_add_import(self):
         ctx.add_import(ctx2_1)
         ctx3 = Context(ident='http://example.com/context_3')
         ctx3.add_import(ctx)
-        final_ctx = Context(ident='http://example.com/context_1', imported=(ctx3,))
+        final_ctx = Context(
+            ident='http://example.com/context_1', imported=(ctx3,))
         self.assertEqual(len(final_ctx.save_imports()), 4)
+
+    def test_init_len(self):
+        ctx = Context(ident='http://example.com/context_1')
+        self.assertEqual(len(ctx), 0)
+
+    def test_len(self):
+        ident_uri = 'http://example.com/context_1'
+        ctx = Context(ident=ident_uri)
+        ctx.add_import(Context(ident=ident_uri))
+        for i in range(0, 5):
+            ctx.add_statement(create_mock_statement(ident_uri, i))
+        self.assertEqual(len(ctx), 5)
+
+    def test_add_remove_statement(self):
+        ident_uri = 'http://example.com/context_1'
+        ctx = Context(ident=ident_uri)
+        stmt1 = create_mock_statement(ident_uri, 1)
+        stmt2 = create_mock_statement(ident_uri, 2)
+        stmt3 = create_mock_statement(ident_uri, 3)
+        stmt4 = create_mock_statement(ident_uri, 4)
+        [ctx.add_statement(stmt) for stmt in [stmt1, stmt2, stmt3, stmt4]]
+        ctx.remove_statement(stmt2)
+        self.assertTrue(len(list(ctx.contents_triples())), 3)
+
+    def test_add_bad_statement(self):
```

Comment:
Can you rename this method to a more descriptive `test_add_statement_with_different_context`? I read "bad" and think it's the wrong type, to short, too long, or something like that.